frontend/src/components/Shared/CrossingMapPage/CrossingMapSidebar.js

```diff
@@ -55,83 +63,99 @@ class CrossingMapSidebar extends Component {
-                    onClick={toggleShowOpen}
-                  />
-                  Open
-                </label>
-                <label className="CrossingMapPage_sidebar-filter">
-                  <input
-                    className="CrossingMapPage_sidebar-filter-checkbox"
-                    type="checkbox"
-                    defaultChecked={showCaution}
-                    onClick={toggleShowCaution}
-                  />
-                  Caution
-                </label>
-                <label className="CrossingMapPage_sidebar-filter">
-                  <input
-                    className="CrossingMapPage_sidebar-filter-checkbox"
-                    type="checkbox"
-                    defaultChecked={showClosed}
-                    onClick={toggleShowClosed}
-                  />
-                  Closed
-                </label>
-                <label className="CrossingMapPage_sidebar-filter">
-                  <input
-                    className="CrossingMapPage_sidebar-filter-checkbox"
-                    type="checkbox"
-                    defaultChecked={showLongterm}
-                    onClick={toggleShowLongterm}
+
+            {!searchFocused && (
+              <div>
+                {selectedCrossingId && (
+                  <SelectedCrossingContainer
+                    crossingId={selectedCrossingId}
+                    currentUser={currentUser}
+                    selectCrossing={selectCrossing}
                   />
-                  Long Term Closure
-                </label>
+                )}
+                <div className="CrossingMapPage_sidebar-filter-sort-toggle-container">
+                  <div
+                    className={classnames(
+                      'CrossingMapPage_sidebar-filter-toggle',
+                      {
+                        selected: this.state.showFilters,
+                      },
+                    )}
+                    onClick={this.toggleFilters}
+                  >
+                    <div className="CrossingMapPage_sidebar-filter-toggle-text">
+                      {this.state.showFilters ? (
+                        <FontAwesome name="minus" ariaLabel="Hide" />
+                      ) : (
+                        <FontAwesome name="plus" ariaLabel="Show" />
+                      )}{' '}
+                      FILTER
+                    </div>
+                  </div>
+                </div>
+                {this.state.showFilters && (
+                  <div className="CrossingMapPage_sidebar-filter-container">
+                    <label className="CrossingMapPage_sidebar-filter">
+                      <input
+                        className="CrossingMapPage_sidebar-filter-checkbox"
+                        type="checkbox"
+                        defaultChecked={showOpen}
+                        onClick={toggleShowOpen}
+                      />
+                      Open
+                    </label>
```

Comment:
It might be helpful to make a checkbox component.

For example:

```js
const FilterCheckbox = ({defaultChecked, onClick, title}) => (
  <label className="CrossingMapPage_sidebar-filter">
    <input
      className="CrossingMapPage_sidebar-filter-checkbox"
      type="checkbox"
      defaultChecked={defaultChecked}
      onClick={}
    />
     {title}
  </label>
)

...
 
<FilterCheckbox
  title="Closed"
  defaultChecked={showClosed}
  onClick={toggleShowClosed}
/>
```

This should make the code here a little easier to read.

Reply:
Good call! I had started down the path of trying to make it so I only needed to do:
`<FilterCheckbox blarg={'closed'}` for each one but that led to way too much logic in a checkbox component, this is a great clean solution.